airflow/api_connexion/security.py

```diff
@@ -35,3 +35,23 @@ def decorated(*args, **kwargs):
         return function(*args, **kwargs)
 
     return cast(T, decorated)
+
+
+def requires_access(permissions: Sequence[Tuple[str, str]]) -> Callable[[T], T]:
+    """
+    Factory for decorator that checks current user's permissions against required permissions.
+    """
+
+    def requires_access_decorator(func: T):
+        @wraps(func)
+        def wrapped_function(*args, **kwargs):
+            appbuilder = current_app.appbuilder
+            for permission in permissions:
+                if not appbuilder.sm.has_access(*permission):
+                    return Response("Forbidden", 403)
```

Comment:
```suggestion
                    raise PermissionDenied("Forbidden", 403)
```
It returns a [Problem Details](https://tools.ietf.org/html/draft-ietf-appsawg-http-problem-00) error response